src/wazuh_db/wdb_integrity.c

```diff
@@ -319,7 +320,7 @@ int wdbi_query_checksum(wdb_t * wdb, wdb_component_t component, const char * com
         switch (retval) {
         case 0: // No data
         case 1: // Checksum failure
-            wdbi_update_attempt(wdb, component, timestamp);
+            wdbi_update_attempt(wdb, component, timestamp, checksum);
```

Comment:
The agent's checksum should be saved for every `integrity_check_global` and not only for failed attempts

Reply:
+1

Reply:
Done